src/k8s-engine/k8s.ts

```diff
@@ -70,11 +83,16 @@ export interface KubernetesBackend extends events.EventEmitter {
    */
   getBackendInvalidReason(): Promise<KubernetesError | null>;
 
+  /**
+   * Create the singleton image-event handler and associate it with the k8s-manager subclass
+   */
+  createImageEventHandler(engineName: string): void;
```

Comment:
Shouldn't this just happen automatically as needed? Why does this require a separate call?

(This is, the k8s backend should have a protected lazy getter to create an image event handler, based on the current engine.)

Reply:
No, having multiple image event-handlers is too complicated, because then we have to constantly tun listeners on and off.  See my previous comment and tech-debt on moving all the image-related event handlers to the image-event handler (this seems so obvious, but I didn't do it in this PR because the image-related events were handled with two separate protocols, one through Event Emitters and one through `main-events`).